assembly/src/assembler/context.rs

```diff
@@ -439,14 +438,17 @@ impl ModuleContext {
     /// compiling a program, the executable module will have the main procedure left on the
     /// procedure stack. To complete the module we need to pop the main procedure off the stack and
     /// append its callset to the callset of the module context.
+    /// 
+    /// # Panics
+    /// Panics if:
+    /// - There is not exactly one module left on the procedure stack.
+    /// - If this module is not an executable module.
+    /// - If the procedure context is not main.
     pub fn complete_executable(&mut self) {
         assert!(self.is_executable(), "module not executable");
-
-        let main_proc_context = self.proc_stack.pop().expect("no procedures");
-
+        assert_eq!(self.proc_stack.len(), 1, "procedure stack should only have the main procedure. proc_stack length should be 1");
+        let main_proc_context = self.proc_stack.pop().unwrap();
         assert!(main_proc_context.is_main(), "not main procedure");
```

Comment:
I wonder if this shouldn't be a case to justify a signature change for the function...Usually, when we can't describe the assertion at the beginning of the function, then chances are we the function shouldn't be infallible.

But, sticking to the scope of the PR, the implementation is fine as it is!